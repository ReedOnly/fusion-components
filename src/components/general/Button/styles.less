@import "~styles/variables.less";
@import "~styles/colors.less";
@import "~styles/fonts.less";

@import "./variables.less";
@import "./variants.less";
@import "./sizes.less";

.container {
    font-family: Equinor;
    display: inline-block;
    background: none;
    border: none;
    outline: none;
    cursor: pointer;
    border-radius: var(--corner-radius);
    box-sizing: border-box;

    & + & {
        margin-left: calc(var(--grid-unit) * 1px) !important;
    }

    &:focus {
        outline: 1px dashed rgba(0, 0, 0, .5);
    }

    &:active {
        outline: none;
    }

    .button {
        display: inline-block;
        box-sizing: border-box;
        border: 1px solid;
        border-radius: var(--corner-radius);
        position: relative;
        z-index: 1;
        min-width: 0;
        vertical-align: middle;
        overflow: hidden;

        label{
            cursor: pointer;
        }
    }

    .icon{
        text-align: center;
        position: absolute;
        top: 0;
        left: 0;
    }

    &.block {
        display: block;
        width: calc(100% + (var(--grid-unit) * 1px));

        .button {
            display: block;
        }
    }
    &.flex {
        display: flex;
        width: calc(100% + (var(--grid-unit) * 1px));

        .button {
            display: flex;
        }
    }

    /* Variants */
    &.contained {
        .variant-contained(primary, @color-primary, @color-primary-text, @color-primary-alt1, @color-primary-pressed, @color-accent3, @color-primary);
        .variant-contained(signal, @color-signal, @color-signal-text, @color-signal-alt1, @color-signal-alt3, @color-signal-text, @color-signal);

        &[disabled] {
            .button-disabled();
        }
    }

    &.outlined {
        .variant-outlined(primary, #243746, @color-primary, @color-primary-pressed, @color-accent3, @color-primary);
        .variant-outlined(signal, #243746, @color-signal, @color-signal-alt3,  @color-signal-text, #243746);

        &[disabled] {
            .button-disabled();
        }
    }

    &.frameless {
        .variant-frameless(primary, @color-primary, @color-accent2, @color-primary-pressed, @color-accent3, @color-primary);
        .variant-frameless(signal, @color-signal, @color-accent1, @color-signal-alt3, @color-signal, @color-signal-text);

        &[disabled] {
            .button-disabled-frameless() !important;
        }
    }

    /* Sizes */
    .button-sizes();
}