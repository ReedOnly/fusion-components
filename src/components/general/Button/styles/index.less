@import "~styles/variables.less";
@import "~styles/colors.less";

@import "./variables.less";
@import "./variants.less";
@import "./sizes.less";

.container {
    display: inline-block;
    background: none;
    border: none;
    outline: none;
    cursor: pointer;
    border-radius: var(--corner-radius);
    box-sizing: border-box;

    & + & {
        margin-left: calc(var(--grid-unit) * 1px) !important;
    }

    &:focus {
        outline: 1px dashed rgba(0, 0, 0, .5);
    }

    &:active {
        outline: none;
    }

    .button {
        display: inline-block;
        box-sizing: border-box;
        border: 1px solid;
        border-radius: var(--corner-radius);
        position: relative;
        z-index: 1;
        min-width: 0;
        vertical-align: middle;
        overflow: hidden;

        label {
            cursor: pointer;
        }
    }

    &.block {
        display: block;
        width: calc(100% + (var(--grid-unit) * 1px));

        .button {
            display: block;
        }
    }

    &.flex {
        display: flex;
        width: calc(100% + (var(--grid-unit) * 1px));

        .button {
            display: flex;
        }
    }

    /* Variants */
    &.contained {
        .variant-contained(primary, @color-primary, @color-primary-text, @color-primary-alt1, @color-primary-pressed);
        .variant-contained(signal, @color-signal, @color-signal-text, @color-signal-alt1, @color-signal-alt3);

        &[disabled] {
            .button-disabled();
        }
    }

    &.outlined {
        .variant-outlined(primary, #243746, @color-primary, @color-primary-pressed);
        .variant-outlined(signal, #243746, @color-signal, @color-signal-alt3);

        &[disabled] {
            .button-disabled();
        }
    }

    &.frameless {
        .variant-frameless(primary, @color-primary, @color-accent2, @color-primary-pressed);
        .variant-frameless(signal, @color-signal, @color-accent1, @color-signal-alt3);

        &[disabled] {
            .button-disabled-frameless() !important;
        }
    }

    /* Sizes */
    .button-sizes();
}