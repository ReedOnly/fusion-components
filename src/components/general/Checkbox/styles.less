@import "~styles/variables.less";
@import "~styles/colors.less";

.checkboxContainer {
  --checkbox-height: calc(var(--grid-unit) * 2px);
  --checkbox-width: calc(var(--grid-unit) * 2px);

  height: calc(var(--grid-unit) * 6px);
  padding: 0 calc((var(--grid-unit) * 2px) - 3px) 0
    calc((var(--grid-unit) * 2px) - 3px);
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  user-select: none;

  &.disabled {
    cursor: not-allowed;
  }

  &.comfortable {
    input + label {
      font-size: 14px;
    }
  }

  &.compact {
    input + label {
      font-size: 11px;
    }
  }

  input {
    opacity: 0;
    height: 0;
    width: 0;

    &:checked ~ .checkmark {
      &::after {
        display: block;
      }
    }

    &:focus ~ .checkmark {
      &::before {
        display: block;
      }
    }
  }

  &.primary {
    color: @color-primary;

    & > .checkmark {
      border-color: @color-primary;
    }

    & > input:checked ~ .checkmark {
      background-color: @color-primary;
    }

    & > input:checked:disabled ~ .checkmark {
      background-color: @color-primary-alt2;
    }

    & > input:disabled ~ .checkmark {
      border-color: @color-primary-alt2;
    }
  }

  &.secondary {
    color: @color-signal;

    & > .checkmark {
      border-color: @color-signal;
    }

    & > input:checked ~ .checkmark {
      background-color: @color-signal;
    }

    & > input:checked:disabled ~ .checkmark {
      background-color: @color-signal-alt2;
    }

    & > input:disabled ~ .checkmark {
      border-color: @color-signal-alt2;
    }
  }

  .checkmark {
    height: var(--checkbox-height);
    width: var(--checkbox-width);
    margin-right: calc(var(--grid-unit) * 1px);
    display: flex;
    justify-content: center;
    align-items: center;
    border: 2px solid;
    border-radius: 2px;
    box-sizing: border-box;

    &:after {
      width: calc(var(--checkbox-width) * 0.25);
      height: calc(var(--checkbox-height) * 0.6);
      margin-bottom: calc(var(--checkbox-height) * 0.13);
      border: solid white;
      border-width: 0 2px 2px 0;
      transform: rotate(45deg);
      content: "";
      display: none;
    }

    &:before {
      width: calc(var(--checkbox-width) + 2px);
      height: calc(var(--checkbox-height) + 2px);
      border: 1px dashed #bfbfbf;
      position: absolute;
      content: "";
      display: none;
    }
  }
}
