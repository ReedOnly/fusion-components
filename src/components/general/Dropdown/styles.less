.dropdownContainer {
    position: absolute;
    top: 100%;
    min-width: 100%;
    margin-top: calc(var(--grid-unit) * 0.5px);
    pointer-events: all;
    background: white;
    border-radius: 4px;

    &.justifyLeft {
        right: 50%;
    }
    &.justifyRight {
        left: 50%;
    }
}