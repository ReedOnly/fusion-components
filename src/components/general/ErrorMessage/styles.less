@import "~styles/variables.less";

.container {
    position: relative;
    width: 100%;
    height: 100%;

    .messageContainer {
        position: absolute;
        top: 0; left: 0; bottom: 0; right: 0;
        display: flex;
        align-content: center;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        background: rgba(255, 255, 255, .8);
        z-index: 999;

        .message {
            font-size: 14px;
            line-break: 24px;
            margin-bottom: calc(var(--grid-unit) * 2px);
            text-align: center;
            p{
                font-weight: 500;
            }
        }
        .icon {
            font-size: 100px;
            color: #cccccc;
        }
    }
}