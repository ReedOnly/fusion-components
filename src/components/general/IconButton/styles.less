.container {
    background: none;
    border: none;
    outline: none;
    margin: 0;
    padding: 0;
    cursor: pointer;
    overflow: visible;

    .iconContainer {
        display: flex;
        align-items: center;
        justify-content: center;
        width: 100%;
        height: 100%;
        box-sizing: border-box;
        border: 1px solid transparent;
        border-radius: 50%;
        color: var(--color-primary);
    }

    &.isToggler .iconContainer {
        border-radius: 1px;
    }

    &:hover {
        .iconContainer {
             background-color: var(--color-black-alt5);
             border-color: var(--color-black-alt4);
        }
    }

    &.comfortable {
        width: calc(var(--grid-unit) * 6px);
        height: calc(var(--grid-unit) * 6px);
    }

    &.compact {
        width: calc(var(--grid-unit) * 4px);
        height: calc(var(--grid-unit) * 4px);
    }

    &.isActive {
        .iconContainer {
            background-color: var(--color-primary);
            border-color: var(--color-primary);
            color: var(--color-white);
        }
    }

    &:active {
        .iconContainer {
            background-color: var(--color-primary-alt4);
            border-color: var(--color-primary-alt4);
            box-shadow: 0 0 0 2px var(--color-primary-alt4);
            color: var(--color-primary);
        }
    }
}