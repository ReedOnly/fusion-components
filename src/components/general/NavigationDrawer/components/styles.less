@import '~styles/mixins.less';

.container {
    width: 100%;
    height: .grid-unit(6px)[];
    background-color: var(--color-white);
    display: flex;
    cursor: pointer;
    color: var(--color-primary-accent);
    flex-direction: column;
    justify-content: center;

    .linkContainer {
        display: flex;
        color: inherit;
        width: .grid-unit(27px)[];
        align-self: flex-end;
        height: 100%;
        justify-content: center;
        flex-direction: column;

        .title {
            align-self: center;
            height: .grid-unit(5px)[];
            width: .grid-unit(27px)[];
            display: flex;
            justify-content: center;
            flex-direction: column;
            border-radius: 4px;
        }
    }

    &:hover{
        background-color: var(--color-primary-hover-alt1);
        color: var(--color-primary-hover);
    }

    
    &.isActive {
        color: var(--color-primary);
    }

    &.menuChild {
        .linkContainer{
            width: .grid-unit(33px)[];
            box-shadow: -1px 0px 0px 0px var(--color-black-alt4);
            .title {
                width: .grid-unit(30px)[];
                padding-left: .grid-unit(2px)[];
            }
        }
        &.isActive {
            background-color: var(--color-primary-alt4);
            .linkContainer{
                box-shadow: -13px 0px 0px -12px var(--color-primary), -1px 0px 0px 0px var(--color-black-alt4);
            }   
        }
        &:active{
            background-color: var(--color-white);
            .linkContainer{
                .title {
                    background-color: var(--color-primary-alt4);
                }
               
            }
        }
    }
}