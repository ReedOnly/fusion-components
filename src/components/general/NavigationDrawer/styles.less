@import '~styles/mixins.less';

.container {
    box-sizing: border-box;
    height: 100%;
    border-right: 2px solid var(--color-black-alt4);
    transition: width .1s;
    overflow-y: auto;
    overflow-x: hidden;
    color: var(--color-primary-accent);
    width: .grid-unit(40px)[];
    flex-shrink: 0;

    .collapseButtonContainer {
        width: .grid-unit(6px)[];
        height: .grid-unit(6px)[];
        border-bottom: 1px solid var(--color-black-alt4);
    }

    &.isCollapsed {
        width: .grid-unit(6px)[];
    }
}