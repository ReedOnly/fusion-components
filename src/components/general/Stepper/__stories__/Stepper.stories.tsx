import * as React from 'react';
import * as styles from './styles.less';

import { storiesOf } from '@storybook/react';
import withFusionStory from '../../../../../.storybook/withFusionStory';
import { Stepper, Step } from '../index';

const StepperStory = () => {
    const handleChange = stepKey => console.log(stepKey);

    return (
        <Stepper forceOrder={true} initialStepKey="fill-in-details" onChange={handleChange}>
            <Step title="Select workspace" stepKey="select-workspace">
                <div className={styles.content}>Select workspace</div>
            </Step>
            <Step title="Select report/dashboard" stepKey="select-report-dashboard">
                <div className={styles.content}>Select a report or dashboard</div>
            </Step>
            <Step title="Fill in details" stepKey="fill-in-details">
                <div className={styles.content}>
                    Specify details about refresh rate, data source etc.
                </div>
            </Step>
            <Step title="Summary" stepKey="summary">
                <div className={styles.content}>Summary</div>
            </Step>
            <Step title="Publish" stepKey="publish" disabled>
                <div className={styles.content}>Publish button</div>
            </Step>
        </Stepper>
    );
};

storiesOf('General|Stepper', module)
    .addParameters({ jest: ['Stepper.stories.jsx'] })
    .addDecorator(withFusionStory('Stepper'))
    .add('Default', () => <StepperStory />);
