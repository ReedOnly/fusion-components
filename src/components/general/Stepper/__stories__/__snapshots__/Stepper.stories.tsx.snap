// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General|Stepper Default 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div>
      <div
        className=""
      >
        <button
          className="container comfortable isDisabled"
          disabled={true}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <span
            className="iconContainer"
          >
            <svg
              height={24}
              style={
                Object {
                  "color": "",
                }
              }
              viewBox="0 0 24 24"
              width={24}
            >
              <path
                clipRule="evenodd"
                d="M18.7912 11.005H7.62124L12.5012 6.12502C12.8912 5.73502 12.8912 5.09502 12.5012 4.70502C12.3144 4.51777 12.0608 4.41254 11.7962 4.41254C11.5317 4.41254 11.2781 4.51777 11.0912 4.70502L4.50124 11.295C4.11124 11.685 4.11124 12.315 4.50124 12.705L11.0912 19.295C11.4812 19.685 12.1112 19.685 12.5012 19.295C12.8912 18.905 12.8912 18.275 12.5012 17.885L7.62124 13.005H18.7912C19.3412 13.005 19.7912 12.555 19.7912 12.005C19.7912 11.455 19.3412 11.005 18.7912 11.005Z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className=""
      >
        <button
          className="container comfortable"
          disabled={false}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <span
            className="iconContainer"
          >
            <svg
              height={24}
              style={
                Object {
                  "color": "",
                }
              }
              viewBox="0 0 24 24"
              width={24}
            >
              <path
                clipRule="evenodd"
                d="M5.20874 13H16.3787L11.4987 17.88C11.1087 18.27 11.1087 18.91 11.4987 19.3C11.8887 19.69 12.5187 19.69 12.9087 19.3L19.4987 12.71C19.8887 12.32 19.8887 11.69 19.4987 11.3L12.9187 4.70002C12.7319 4.51276 12.4783 4.40753 12.2137 4.40753C11.9492 4.40753 11.6956 4.51276 11.5087 4.70002C11.1187 5.09002 11.1187 5.72002 11.5087 6.11002L16.3787 11H5.20874C4.65874 11 4.20874 11.45 4.20874 12C4.20874 12.55 4.65874 13 5.20874 13Z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
      <div>
        <div>
          <a
            className=""
            onClick={[Function]}
          >
            <div
              className=""
            >
              1
            </div>
            <div
              className=""
            >
              <span>
                 
                Select workspace and do some other work
              </span>
              <span />
            </div>
            <span />
            <span>
              This is a description text
            </span>
          </a>
          <a
            className=""
            onClick={[Function]}
          >
            <div
              className=""
            >
              2
            </div>
            <div
              className=""
            >
              <span>
                 
                Select report/dashboard
              </span>
              <span />
            </div>
            <span />
            <span>
              Description
            </span>
          </a>
          <a
            className=""
            onClick={[Function]}
          >
            <div
              className=""
            >
              3
            </div>
            <div
              className=""
            >
              <span>
                 
                Fill in details
              </span>
              <span />
            </div>
            <span />
            <span />
          </a>
          <a
            className=""
            onClick={[Function]}
          >
            <div
              className=""
            >
              4
            </div>
            <div
              className=""
            >
              <span>
                 
                Summary
              </span>
              <span />
            </div>
            <span />
            <span />
          </a>
          <span
            className="undefined"
          >
            <div
              className=""
            >
              5
            </div>
            <div
              className="undefined"
            >
              <span>
                Publish
              </span>
            </div>
            <span />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`General|Stepper Interactive 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div>
      <div
        className=""
      >
        <button
          className="container comfortable isDisabled"
          disabled={true}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <span
            className="iconContainer"
          >
            <svg
              height={24}
              style={
                Object {
                  "color": "",
                }
              }
              viewBox="0 0 24 24"
              width={24}
            >
              <path
                clipRule="evenodd"
                d="M18.7912 11.005H7.62124L12.5012 6.12502C12.8912 5.73502 12.8912 5.09502 12.5012 4.70502C12.3144 4.51777 12.0608 4.41254 11.7962 4.41254C11.5317 4.41254 11.2781 4.51777 11.0912 4.70502L4.50124 11.295C4.11124 11.685 4.11124 12.315 4.50124 12.705L11.0912 19.295C11.4812 19.685 12.1112 19.685 12.5012 19.295C12.8912 18.905 12.8912 18.275 12.5012 17.885L7.62124 13.005H18.7912C19.3412 13.005 19.7912 12.555 19.7912 12.005C19.7912 11.455 19.3412 11.005 18.7912 11.005Z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className=""
      >
        <button
          className="container comfortable isDisabled"
          disabled={true}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <span
            className="iconContainer"
          >
            <svg
              height={24}
              style={
                Object {
                  "color": "",
                }
              }
              viewBox="0 0 24 24"
              width={24}
            >
              <path
                clipRule="evenodd"
                d="M5.20874 13H16.3787L11.4987 17.88C11.1087 18.27 11.1087 18.91 11.4987 19.3C11.8887 19.69 12.5187 19.69 12.9087 19.3L19.4987 12.71C19.8887 12.32 19.8887 11.69 19.4987 11.3L12.9187 4.70002C12.7319 4.51276 12.4783 4.40753 12.2137 4.40753C11.9492 4.40753 11.6956 4.51276 11.5087 4.70002C11.1187 5.09002 11.1187 5.72002 11.5087 6.11002L16.3787 11H5.20874C4.65874 11 4.20874 11.45 4.20874 12C4.20874 12.55 4.65874 13 5.20874 13Z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
      <div>
        <div>
          <a
            className=""
            onClick={[Function]}
          >
            <div
              className=""
            >
              1
            </div>
            <div
              className=""
            >
              <span>
                 
                Select workspace
              </span>
              <span />
            </div>
            <span />
            <span />
          </a>
          <span
            className="undefined"
          >
            <div
              className=""
            >
              2
            </div>
            <div
              className=""
            >
              <span>
                Select report/dashboard
              </span>
            </div>
            <span>
              Description
            </span>
          </span>
          <span
            className="undefined"
          >
            <div
              className=""
            >
              3
            </div>
            <div
              className=""
            >
              <span>
                Fill in details
              </span>
            </div>
            <span />
          </span>
          <span
            className="undefined"
          >
            <div
              className=""
            >
              4
            </div>
            <div
              className=""
            >
              <span>
                Summary
              </span>
            </div>
            <span />
          </span>
          <span
            className="undefined"
          >
            <div
              className=""
            >
              5
            </div>
            <div
              className="undefined"
            >
              <span>
                Publish
              </span>
            </div>
            <span />
          </span>
        </div>
      </div>
    </div>
    <div>
      <div>
        Press the button to enable navigation to next step.
      </div>
      <div>
        <button
          className="container contained primary comfortable"
          disabled={false}
          onClick={[Function]}
          onClickCapture={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <span
            className="button"
          >
            Do work
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
